### PR TITLE
Rest API test

### PR DESCRIPTION
**Pull request by pgcrooks**
Initial REST API test framework. This branch should remain for future test development.

Currently the only tests are to get Tags, Questions and Rooms and as such needs a lot of further work. 

See #60

### CODE REVIEW

**Comment by pgcrooks** [comment]
This push has one gaping problem - we have to disable Permissions on the REST API. It's up to the reviewers whether they would rather get the tests in quickly, or let me fix the permissions issue first.

File: quiz/settings.py [hunk 1]
```diff
@@ -45,9 +45,7 @@
 )
 
 REST_FRAMEWORK = {
-    'DEFAULT_PERMISSION_CLASSES': [
-        'rest_framework.permissions.IsAdminUser'
-    ],
```
Comment (christopherkane) [inline quiz/settings.py]: This is fine, for now. I would rather get functional tests in place, over REST authentication. Can you create another task to track test permissions?

File: quiz/test_rest.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+                 'url':'http://testserver/api/question/{}/'.format(last_pk)})
+
+    def test_get_all_questions(self):
+        """
+        Get all questions
+        """
+        response = self.client.get('/api/question/')
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data['count'], 1)
+
```
Comment (christopherkane) [inline quiz/test_rest.py]: Get a question that doesn't exist?

File: quiz/test_rest.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data, {'name':'test_room', 'code':'{}'.format(Room.objects.last().code)})
+
+    def test_get_all_rooms(self):
+        """
+        Get all rooms
+        """
+        response = self.client.get('/api/room/')
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data['count'], 1)
```
Comment (christopherkane) [inline quiz/test_rest.py]: Get a room that doesn't exist? Much like the tag test?

File: quiz/settings.py (hunk 1 above)
Comment (pgcrooks) [reply quiz/settings.py]: Sure, I'll create one in my name!

**Comment by pgcrooks** [comment]
Thanks @christopherkane , I will add those additional tests in.

**Pull request merged**